tests/api.c

```diff
@@ -1042,33 +1042,63 @@ struct RealPathTestCase realPathTestCases[] = {
     { "/home/C:/ok", "/home/C:/ok" },
     { "/home/fred/frob/frizz/../../../barney/bar/baz/./././../..",
         "/home/barney" },
+    { "/home/fred/sample.", "/home/fred/sample." },
+    { "/home/fred/sample.jpg", "/home/fred/sample.jpg" },
+    { "/home/fred/sample./other", "/home/fred/sample./other" },
+    { "/home/fred/sample.dir/other", "/home/fred/sample.dir/other" },
+    { "./sample.", "/C:/Users/fred/sample." },
+    { "./sample.jpg", "/C:/Users/fred/sample.jpg" },
+    { "./sample./other", "/C:/Users/fred/sample./other" },
+    { "./sample.dir/other", "/C:/Users/fred/sample.dir/other" },
+    { "\\C:\\Users\\fred\\Documents\\junk.txt",
+        "/C:/Users/fred/Documents/junk.txt" },
+    { "C:\\Users\\fred\\Documents\\junk.txt",
+        "/C:/Users/fred/Documents/junk.txt" },
+    { "/C:\\Users\\fred/Documents\\junk.txt",
+        "/C:/Users/fred/Documents/junk.txt" },
 };
-const char* defaultPath = "/C:/Users/fred";
 
-static void test_wolfSSH_RealPath(void)
+struct RealPathTestCase realPathNull[] = {
+    { ".", "/" },
+    { "", "/" },
+    { "..", "/" },
+    { "../barney", "/barney" },
+};
+
+static void DoRealPathTestCase(const char* path, struct RealPathTestCase* tc)
 {
-    struct RealPathTestCase* tc;
     char testPath[128];
     char checkPath[128];
-    word32 testCount =
-        (sizeof realPathTestCases)/(sizeof(struct RealPathTestCase));
-    word32 i;
     int err;
 
-    for (i = 0, tc = realPathTestCases; i < testCount; i++, tc++) {
-        WSTRNCPY(testPath, tc->in, sizeof(testPath) - 1);
-        testPath[sizeof(testPath) - 1] = 0;
-        WMEMSET(checkPath, 0, sizeof checkPath);
-        err = wolfSSH_RealPath(defaultPath, defaultPath, testPath,
-                checkPath, sizeof checkPath);
-        if (err || WSTRCMP(tc->exp, checkPath) != 0) {
-            printf("RealPath failure (case %u: %d)\n"
-                   "    defaultPath: %s\n"
-                   "          input: %s\n"
-                   "       expected: %s\n"
-                   "         output: %s\n", i, err,
-                   defaultPath, tc->in, tc->exp, checkPath);
-        }
+    WSTRNCPY(testPath, tc->in, sizeof(testPath) - 1);
+    testPath[sizeof(testPath) - 1] = 0;
+    WMEMSET(checkPath, 0, sizeof checkPath);
+    err = wolfSSH_RealPath(path, testPath,
+            checkPath, sizeof checkPath);
+    if (err || WSTRCMP(tc->exp, checkPath) != 0) {
+        printf("RealPath failure (%d)\n"
```

Comment:
fprintf to stderr preferred for diagnostics and error messages.

Reply:
Changing it.